impl/sql/mutationstorage/mutation_logs.go

```diff
@@ -188,6 +190,10 @@ func (m *Mutations) HighWatermark(ctx context.Context, directoryID string, logID
 // ReadLog may return more rows than batchSize in order to fetch all the rows at a particular timestamp.
 func (m *Mutations) ReadLog(ctx context.Context, directoryID string,
 	logID int64, low, high time.Time, batchSize int32) ([]*mutator.LogMessage, error) {
+	// Advance the low and high marks to the next highest quantum to preserve read semantics.
+	low = low.Add(quantum - 1).Truncate(quantum)
```

Comment:
If you don't do this, what changes? I would expect >= to do the right thing...

Reply:
The storage implementation will truncate low to the precision of quantum, with the effect that the timestamp will actually go backwards.